homeassistant/components/onewire/config_flow.py

```diff
@@ -164,3 +177,163 @@ async def async_step_mount_dir(
+        self.configurable_devices = {
+            self._get_device_long_name(device_registry, device.id): device
+            for device in all_devices
+            if device.family in DEVICE_SUPPORT_OPTIONS
+        }
+
+        return await self.async_step_device_selection(user_input=None)
+
+    async def async_step_device_selection(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Select what devices to configure."""
+        errors = {}
+        if user_input is not None:
+            if user_input.get(INPUT_ENTRY_CLEAR_OPTIONS):
+                # Reset all options
+                self.options = {}
+                return await self._update_options()
+
+            selected_devices: list[str] = (
+                user_input.get(INPUT_ENTRY_DEVICE_SELECTION) or []
+            )
+            if selected_devices:
+                self.devices_to_configure = {
+                    device_name: self.configurable_devices[device_name]
+                    for device_name in selected_devices
+                }
+
+                return await self.async_step_configure_device(user_input=None)
+            errors["base"] = "device_not_selected"
+
+        return self.async_show_form(
+            step_id="device_selection",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        INPUT_ENTRY_CLEAR_OPTIONS,
+                        default=False,
+                    ): bool,
+                    vol.Optional(
+                        INPUT_ENTRY_DEVICE_SELECTION,
+                        default=self._get_current_configured_sensors(),
+                        description="Multiselect with list of devices to choose from",
+                    ): cv.multi_select(
+                        {device: False for device in self.configurable_devices.keys()}
+                    ),
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_configure_device(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Config precision option for device."""
+        if user_input is not None:
+            self._update_device_options(user_input)
+            if self.devices_to_configure:
+                return await self.async_step_configure_device(user_input=None)
+            return await self._update_options()
+
+        self.current_device, description = self.devices_to_configure.popitem()
+        data_schema: vol.Schema
+        if description.family == "28":
+            data_schema = vol.Schema(
+                {
+                    vol.Required(
+                        OPTION_ENTRY_SENSOR_PRECISION,
+                        default=self._get_current_setting(
+                            description.id, OPTION_ENTRY_SENSOR_PRECISION, "Default"
+                        ),
+                    ): vol.In(list(PRECISION_MAPPING_FAMILY_28.keys())),
```

Comment:
Fixed!
(Removed call to keys function)

Reply:
You can still improve it by passing a dict

Reply:
Ok, but that would render e.g. "temperature9" instead of "9 Bits". So I would have to change the dict (like switch key<--->value), right?

Reply:
correct.

Reply:
Then, in my opinion this would not be an improvement. While the PRECISION_MAPPING_FAMILY_28 dictionary happens to be bijective this is not necessarily the case for other config mappings.

Reply:
The schemas will be device specific, so `other config mappings` don't matter.
I think I agree with @balloob on this one, and that it's ok to invert the dict and store `temperature` / `temperature9` in the config instead of `Default` / `9 bit`.

Reply:
I agree that storing the friendly name string "9 Bit" in the config is a bad choice (because of the space and uppercase letters etc. ). I don't think that the keyword that is used in the server-url should be used in the config. The user/configurer should not have to know about these details. Even though this config is not hand written I think that "9bits" is better in the config storage then "temperature9", and both are better than "9 Bits".

So, I guess what I am suggesting is having all three of them in some sort of map. Config option, friendly name, and server-url. However, you make the call and I will abide :)

Reply:
For device family 28, I think the server URL is the best (and easiest) choice. I'm not sure why you want to introduce a distinct "config option" name.

Reply:
> I'm not sure why you want to introduce a distinct "config option" name.
I will drop the subject after this, but just to answer your question: Because "temperature9" does not mean anything to anyone who does not now about the server and the url while "9bits" do.